src/librustdoc/html/static/storage.js

```diff
@@ -89,35 +89,20 @@ function hasOwnProperty(obj, property) {
     return Object.prototype.hasOwnProperty.call(obj, property);
 }
 
-function usableLocalStorage() {
-    // Check if the browser supports localStorage at all:
-    if (typeof Storage === "undefined") {
-        return false;
-    }
-    // Check if we can access it; this access will fail if the browser
-    // preferences deny access to localStorage, e.g., to prevent storage of
-    // "cookies" (or cookie-likes, as is the case here).
-    try {
-        return window.localStorage !== null && window.localStorage !== undefined;
-    } catch(err) {
-        // Storage is supported, but browser preferences deny access to it.
-        return false;
-    }
-}
-
 function updateLocalStorage(name, value) {
-    if (usableLocalStorage()) {
-        localStorage[name] = value;
-    } else {
-        // No Web Storage support so we do nothing
+    try {
+        window.localStorage.setItem(name, value);
+    } catch(e) {
+        // localStorage is not accessible, do nothing
```

Comment:
Wouldn't it make sense to attempt to call `localStorage.clear()` here in case enough data accumulated to reach the per-site limit?

Reply:
Why would you want to remove all your data if you can't add more?

Reply:
That's assuming something filled up the storage to its limit which prevents storing new data, e.g. due to previous iterations of the site.

Reply:
We store very little amount of information in the local storage (less than 40 bytes iirc), so we should be fine. ;)

Reply:
Maybe this can be discussed outside of the context of this pull request ?
If new behavior needs to be introduced, it can be independently of this bug fix.

Reply:
No need, if there's nothing that could cause unbounded growth then the limitation is probably due to something else, e.g. a browser simply having localstorage disabled, `clear()` wouldn't help with that.

Reply:
Are we good to merge, then ?

Reply:
I already approved it so it should be merged in the next days.